nosql-db/elasticsearch/src/test/java/io/quarkus/ts/elasticsearch/ElasticsearchIT.java

```diff
@@ -0,0 +1,126 @@
+import io.quarkus.test.services.Container;
+import io.quarkus.test.services.QuarkusApplication;
+import io.quarkus.test.services.URILike;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import io.vertx.core.json.JsonObject;
+
+@QuarkusScenario
+public class ElasticsearchIT {
+
+    static final int ELASTIC_PORT = 9200;
+
+    @Container(image = "${elastic.7x.image}", port = ELASTIC_PORT, expectedLog = "started")
+    static DefaultService elastic = new DefaultService()
+            .withProperty("discovery.type", "single-node")
+            .withProperty("ES_JAVA_OPTS", "-Xms512m -Xmx512m")
+            .withProperty("cluster.routing.allocation.disk.threshold_enabled", "false")
+            .withProperty("xpack.security.enabled", "false");
+
+    @QuarkusApplication
+    static RestService app = new RestService()
+            .withProperty("quarkus.elasticsearch.hosts", () -> getElasticSearchConnectionChain(elastic.getURI(Protocol.HTTP)));
+
+    protected static String getElasticSearchConnectionChain(URILike uri) {
+        return uri.toString().replaceAll("http://", "");
+    }
+
+    protected RestService getApp() {
+        return app;
+    }
+
+    @Test
+    void dataCreatedAndAccessible() {
+        // CREATE content
+        given()
+                .contentType(ContentType.JSON)
+                .body("{\"name\": \"apples\", \"color\": \"green\"}")
+                .when()
+                .post("fruits")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+
+        // SEARCH content
+        final StringBuilder sb = new StringBuilder();
+        // Elasticsearch is creating index and mapping, thus await loop
+        await().pollInterval(500, TimeUnit.MILLISECONDS).atMost(10, TimeUnit.SECONDS).untilAsserted(() -> {
+            String id = given()
+                    .when()
+                    .get("fruits/search?color=green")
+                    .then()
+                    .statusCode(HttpStatus.SC_OK)
+                    .body(containsString("apples"))
+                    .extract().path("[0].id");
+            sb.append(id);
+        });
+
+        // DELETE content
+        given()
+                .when()
+                .delete("fruits/" + sb)
+                .then()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+        // wait for Elasticsearch to finish the work
+        await().pollInterval(500, TimeUnit.MILLISECONDS).atMost(10, TimeUnit.SECONDS).untilAsserted(() -> {
+            given()
+                    .when()
+                    .get("fruits/search?color=green")
+                    .then()
+                    .statusCode(HttpStatus.SC_OK)
+                    .body("isEmpty()", is(true));
+        });
+    }
+
+    @Test
+    void dataTypesCheck() {
+        DataTypes fooDataTypes = new DataTypes();
+        fooDataTypes.id = UUID.randomUUID().toString();
+        fooDataTypes.name = "Foo name ěščřžýáíéůú ťďň";
```

Comment:
Optional:

TP says it will test stored texts with special chars like diacritics, but we only store it and retrieve it based on id, so we don't test searching for texts with diacritics. I think it should be fine as query should be encoded / decoded in a same fashion (same request class?), but normally we would check?

https://github.com/quarkus-qe/quarkus-test-plans/commit/ca7028f7ecf0bafca9f1574a1597037832ba769f#diff-e8c13bb910976634c4b0cf2bc80fa3e6ed414373d6c8f213179f3cdec30afebeR15

Reply:
PR updated